apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -1,47 +1,184 @@
-/*global BridgeServiceMixin */
+/*global bridge,
+         BridgeServiceMixin,
+         BroadcastChannel,
+         Draft,
+         Drafts,
+         performance,
+         streamClient
+*/
 'use strict';
 
-if (!('BridgeServiceMixin' in self)) {
-  importScripts('/services/js/bridge_service_mixin.js');
-}
+/**
+ * Type description for the ConversationSummary object that is return from the
+ * Conversation service.
+ * @typedef {Object} ConversationSummary
+ * @property {number} id Unique identifier of the the conversation.
+ * @property {string} body Short content of the last message in the
+ * conversation.
+ * @property {Array.<string>} participants List of the conversation
+ * participants.
+ * @property {Date} timestamp Conversation timestamp.
+ * @property {{ hasUnread: boolean, hasNewError: boolean }} status Conversation
+ * status that indicates whether conversation has unread messages or/and failed
+ * messages that user hasn't seen yet.
+ * @property {string} lastMessageType Type of the last message in the
+ * conversation. Can be either 'sms' or 'mms'.
+ * @property {Draft} draft If conversation is draft or has draft, then this
+ * field contains appropriate draft information, otherwise it's null.
+ */
+
+[
+  ['bridge', '/lib/bridge/bridge.js'],
+  ['streamClient', '/lib/bridge/plugins/stream/client.js'],
+  ['BridgeServiceMixin', '/services/js/bridge_service_mixin.js'],
+  ['Drafts', '/services/js/drafts.js']
+].forEach(([dependencyName, dependencyPath]) => {
+  if (!(dependencyName in self)) {
+    importScripts(dependencyPath);
+  }
+});
 
 (function(exports) {
-  const SERVICE_NAME = 'conversation-service';
+  const debug = 0 ?
+    (arg1, ...args) => console.log(`[ConversationService] ${arg1}`, ...args):
+    () => {};
 
-  const STREAMS = Object.freeze(
-    ['getAllConversations', 'getMessagesForConversation']
-  );
+  const mark = 0 ?
+    (markName) =>
+      performance.mark(`[ConversationService] ${markName}@sms.gaiamobile.org`):
+    () => {};
+
+  const priv = {
+    mobileMessageClients: Symbol('mobileMessageClients'),
+
+    getMobileMessageClient: Symbol('getMobileMessageClient')
+  };
+
+  const SERVICE_CONTRACT = Object.freeze({
+    name: 'conversation-service',
+
+    streams: Object.freeze(
+      ['getAllConversations', 'getMessagesForConversation']
+    ),
+
+    methods: Object.freeze([
+      'deleteConversations', 'deleteMessages', 'markConversationsAs',
+      'getConversationSummary', 'getMessage', 'findConversationFromAddress'
+    ]),
 
-  const METHODS = Object.freeze([
-    'deleteConversations', 'deleteMessages', 'markConversationsAs',
-    'getConversationSummary', 'getMessage', 'findConversationFromAddress'
-  ]);
+    events: Object.freeze(['message-change'])
+  });
 
-  const EVENTS = Object.freeze([
-    'message-change'
-  ]);
+  function draftToConversationSummary(draft) {
```

Comment:
There are two methods that similar to what we do in Threads.js - but they still slightly different and I didn't want to include Threads.js here. I was going to decide later in follow-ups. Please, tell me if you're ok with this :)